app/models/event.rb

```diff
@@ -35,11 +40,21 @@ def trigger!
   end
 
   private
+  def create_events_groups!
+    self.events_groups.import(
+      related_groups.map {|g| EventsGroup.new(group_id: g.id, event_id: self.id) }
+    )
+  end
 
-  # which communities should know about this event?
-  # Polls override this to look at the communities associated with the poll.
-  def communities
-    Array(eventable&.group&.community)
+  # which groups are privvy to this event?
+  def related_groups
```

Comment:
also I don't know if we need to include self.discussion here. seems like eventable will always be aware of the groups it belongs to.

group or groups thing could be improved i'm sure.

Reply:
I think I'd like any of Poll, Discussion, or Group to respond to a `related_groups` method, which returns an array of groups we care about for that model.

```
module HasGuestGroup
  def related_groups
    [guest_group, group].compact
  end
end
```

```
class Group
  def related_groups
    Array(self)
  end
end
```
```
class Membership
  delegate :related_groups, to: :group
end
```

then

```
class Event
  def create_event_groups!
    self.event_groups.import(
      self.eventable.related_groups.map { |group| EventGroup.new(group: group) }
    )
  end
end
```

Reply:
Sounds cool.. q: why not just call it groups? rather than related_groups? Seems like that's the natural direction to me.